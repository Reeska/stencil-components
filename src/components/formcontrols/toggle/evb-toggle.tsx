import { Component, ComponentInterface, Element, Event, EventEmitter, Prop } from '@stencil/core';

@Component({
  tag: 'evb-toggle',
  shadow: true,
  styleUrl: 'evb-toggle.css'
})
export class EvbToggle implements ComponentInterface {

  @Prop() type: 'default' | 'flat' | 'rotate' = 'default';
  @Prop() labelon = 'On';
  @Prop() labeloff = 'Off';
  @Prop() value: boolean;

  @Event() evbBlur: EventEmitter<void>;
  @Event() evbChange: EventEmitter<boolean>;
  @Event() evbFocus: EventEmitter<void>;
  @Event() evbInput: EventEmitter<boolean>;

  @Element() host: HTMLElement;

  get inputValue(): boolean {
    const input: HTMLInputElement = this.host.shadowRoot.querySelector('input[type=checkbox]');
    return input.checked;
  }

  render() {
    const id = 'toggle';

    const attributes = {
      class: `toggle toggle-${this.type}`,
      checked: this.value,
      id
    };

    const labelAttrs = {
      'data-on': this.labelon,
      'data-off': this.labeloff,
      for: id
    };

    return ([
      <input id='toggle1' {...attributes} type='checkbox'
        onBlur={() => this.evbBlur.emit()}
        onChange={() => this.evbChange.emit(this.inputValue)}
        onFocus={() => this.evbFocus.emit()}
        onInput={() => this.evbInput.emit(this.inputValue)} />,
      <label {...labelAttrs}></label>
    ]);
  }
}
