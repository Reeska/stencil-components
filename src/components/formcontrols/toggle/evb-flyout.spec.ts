import { EvbToggle } from './evb-toggle';

describe('EvbToggle', () => {
  let component: EvbToggle;

  beforeEach(() => component = new EvbToggle());

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(component.render()).toMatchSnapshot();
  });
});
