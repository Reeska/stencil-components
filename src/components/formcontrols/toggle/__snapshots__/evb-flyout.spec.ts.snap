// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvbToggle should create 1`] = `
Array [
  Object {
    "$attrs$": Object {
      "checked": undefined,
      "class": "toggle toggle-default",
      "id": "toggle",
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
      "onInput": [Function],
      "type": "checkbox",
    },
    "$children$": null,
    "$elm$": undefined,
    "$flags$": 0,
    "$key$": undefined,
    "$name$": undefined,
    "$tag$": "input",
  },
  Object {
    "$attrs$": Object {
      "data-off": "Off",
      "data-on": "On",
      "for": "toggle",
    },
    "$children$": null,
    "$elm$": undefined,
    "$flags$": 0,
    "$key$": undefined,
    "$name$": undefined,
    "$tag$": "label",
  },
]
`;
